src/main/resources/assets/js/app/wizard/page/contextwindow/inspect/page/PageTemplateAndControllerSelector.ts

```diff
@@ -176,17 +178,29 @@ export class PageTemplateAndControllerSelector
             this.loadPageTemplates(),
             this.loadPageControllers()
         ]).spread((templateOptions: Option<PageTemplateOption>[], controllerOptions: Option<PageControllerOption>[]) => {
-            const selectedValue: string = this.getValue();
-            this.removeAllOptions();
-            this.initOptionsList(templateOptions, controllerOptions);
-            if (selectedValue) {
-                this.setValue(selectedValue, true);
-            } else {
-                this.selectInitialOption();
-            }
+            this.handleReloaded(templateOptions, controllerOptions);
         }).catch(DefaultErrorHandler.handle);
     }
 
+    private handleReloaded(templateOptions: Option<PageTemplateOption>[], controllerOptions: Option<PageControllerOption>[]) {
+        const valueToSelectOnReload: string = this.getValueToSelectOnReload();
+        this.removeAllOptions();
+        this.initOptionsList(templateOptions, controllerOptions);
+        if (valueToSelectOnReload) {
+            const isAlreadySelected: boolean = !!this.getSelectedOption() && this.getSelectedOption().value === valueToSelectOnReload;
+            if (!isAlreadySelected) {
+                this.setValue(valueToSelectOnReload, true);
```

Comment:
Shouldn't we check that `preselectedValue` is still in the list of options after reload?

Reply:
No need, if option is missing nothing will happen